test/functional/http/get/blocks.js

```diff
@@ -4,405 +4,402 @@ var node = require('../../../node.js');
-				done();
+			return getBlocksPromise(params).then(function (res) {
+				node.expect(res).to.have.property('message').to.equal('Value ' + height + ' is less than minimum 1');
+				return getJsonForKeyPromise(url + params.join('&')).then(function (response) {
+					node.expect(response).to.eql(null);
+				});
 			});
 		});
-	});
 
-	describe('GET /api/blocks/getFees', function () {
-		it('should be ok', function (done) {
-			http.get('/api/blocks/getFees', function (err, res) {
-				node.expect(res.body).to.have.property('success').to.be.ok;
-				node.expect(res.body).to.have.property('fees');
-				node.expect(res.body.fees.send).to.equal(node.fees.transactionFee);
-				node.expect(res.body.fees.vote).to.equal(node.fees.voteFee);
-				node.expect(res.body.fees.dappRegistration).to.equal(node.fees.dappRegistrationFee);
-				node.expect(res.body.fees.dappDeposit).to.equal(node.fees.dappDepositFee);
-				node.expect(res.body.fees.dappWithdrawal).to.equal(node.fees.dappWithdrawalFee);
-				node.expect(res.body.fees.secondSignature).to.equal(node.fees.secondPasswordFee);
-				node.expect(res.body.fees.delegate).to.equal(node.fees.delegateRegistrationFee);
-				node.expect(res.body.fees.multisignature).to.equal(node.fees.multisignatureRegistrationFee);
-				done();
-			});
+		it('should remove entry from cache on new block', function () {
+			var auxResponse;
+			var params = [
+				'height=' + block.blockHeight
+			];
+			
+			return getBlocksPromise(params)
+				.then(function (res) {
+					node.expect(res).to.have.property('blocks').that.is.an('array');
+					auxResponse = res;
+					return getJsonForKeyPromise(url + params.join('&'));
+				})
+				.then(function (response) {
+					node.expect(auxResponse).to.eql(response);
+					return onNewBlockPromise();
+				})
+				.then(function () {
+					return getJsonForKeyPromise(url + params.join('&'));
+				})
+				.then(function (result) {
+					node.expect(result).to.eql(null);
+				});
 		});
 	});
 
-	describe('/getNethash', function () {
+	describe('/', function () {
 
-		it('should be ok', function (done) {
-			http.get('/api/blocks/getNethash', function (err, res) {
-				node.expect(res.body).to.have.property('success').to.be.ok;
-				node.expect(res.body).to.have.property('nethash').to.be.a('string');
-				node.expect(res.body.nethash).to.equal(node.config.nethash);
-				done();
-			});
-		});
-	});
+		describe('id', function () {
 
-	describe('/getMilestone', function () {
+			it('using genesisblock id should return the result', function () {
+				var id = '6524861224470851795';
+				var params = [
+					'id=' + id
+				];
 
-		it('should be ok', function (done) {
-			http.get('/api/blocks/getMilestone', function (err, res) {
-				node.expect(res.body).to.have.property('milestone').to.be.a('number');
-				done();
+				return getBlocksPromise(params).then(function (res){
+					node.expect(res).to.have.property('blocks').to.be.an('array').that.have.nested.property('0.id').equal(id);
```

Comment:
You aren't checking if the status code is being set properly?

Reply:
I will test it when the responses from promises will be changed.